### PR TITLE
Remember previous save/load location

### LINKED ISSUE #3989: Remember previous save/load location
I keep all my saves in my Dropbox folder so I can play them cross-computer, however the Open / Save dialogues always point to the OpenRCT2 saves folder by default. It would be nice if the last accessed folder could be remembered, so that I don't have to click through a ton of folders to get to my saves.

### PR DESCRIPTION
**Pull request by SijmenSchoon**
Fixes #3989.

Works for games, landscapes, scenarios and tracks (but this is commented out due to tracks not being separated or something?)

Also updates my entry in contributors.md

### CODE REVIEW

**Comment by SijmenSchoon** [comment]
Oops, let me fix this

File: src/util/util.c [hunk 1]
```diff
@@ -62,6 +62,21 @@ bool filename_valid_characters(const utf8 *filename)
 	return true;
 }
 
+utf8 *path_get_directory(const utf8 *path)
```
Comment (IntelOrca) [inline src/util/util.c]: `const utf8 *path`

**Comment by SijmenSchoon** [comment]
Seems like making it const broke more than it fixed

File: src/util/util.c [hunk 2]
```diff
@@ -62,7 +62,7 @@ bool filename_valid_characters(const utf8 *filename)
 	return true;
 }
 
-utf8 *path_get_directory(const utf8 *path)
+const utf8 *path_get_directory(const utf8 *path)
```
Comment (IntelOrca) [inline src/util/util.c]: You can't make the result `const` as you are expecting the callee to free it.

Comment (SijmenSchoon) [inline src/util/util.c]: But what did you mean with the note 3 hours ago then?

Comment (IntelOrca) [inline src/util/util.c]: Sorry, I was being an idiot. I thought it didn't say const before the path.

**Comment by IntelOrca** [comment]
I am just quickly testing it, and then I'll merge.

**Comment by IntelOrca** [comment]
Its merged:
https://github.com/OpenRCT2/OpenRCT2/commit/f866e6fac2b88fb8f132469bc6ec68bfb25f504b

**Pull request closed without merging**